packages/grafana-ui/src/components/Forms/Checkbox.tsx

```diff
@@ -178,11 +185,11 @@ export const getCheckboxStyles = stylesFactory((theme: GrafanaTheme2) => {
       height: ${theme.spacing(checkboxSize)};
       border-radius: ${theme.shape.borderRadius()};
       background: ${theme.components.input.background};
-      border: 1px solid ${theme.components.input.borderColor};
+      border: 1px solid ${invalid ? theme.colors.error.border : theme.components.input.borderColor};
 
       &:hover {
         cursor: pointer;
-        border-color: ${theme.components.input.borderHover};
+        border-color: ${invalid ? theme.colors.error.border : theme.components.input.borderHover};
```

Comment:
I like this better, but just for the sake of not having to repeat this tertiary everywhere, we could make a function `getBorderColor` which receives a borderColor and either returns the error one if invalid or the one received otherwise?